React Toolbar content and feedback updates

nicolethoen opened the pull request:
View preview at https://www.katacoda.com/nthoen/courses/react-components/toolbar-filter

Note: event handlers are not firing until https://github.com/patternfly/training-scenarios/issues/210 gets resolved.

abigaeljamie commented on react-components/toolbar-filter/intro.md:
```diff
@@ -1 +1,3 @@
\ No newline at end of file
+In this tutorial, build a toolbar filter and attach it to a table.
+
+Toolbars can be included to allow a user to manage and manipulate a data set. Data can be presented in any valid presentation, a table, a list, or a data visualization (chart), for example. The PatternFly Toolbar component is a flexible layout system that accommodates a variety of configurations that can be adapted to numerous specific needs.
```
```suggestion
Toolbars allow users to manage and manipulate a data set. Data can be presented in any valid presentation, a table, a list, or a data visualization (chart), for example. The PatternFly Toolbar component is a flexible layout system that accommodates a variety of configurations that can be adapted to numerous needs.
```

abigaeljamie commented on react-components/toolbar-filter/step1.md:
```diff
@@ -1,4 +1,69 @@
-Toolbars can be included to allow a user to manage and manipulate a data set. Data can be presented in any valid presentation, a table, a list, or a data visualization (chart), for example. The PatternFly Toolbar component is a flexible layout system that accommodates a variety of configurations that can be adapted to your specific needs.
+At its most basic, Toolbars consist of content rows of components. Most toolbars will only be a single row, but each row needs to be housed in a `DataToolbarContent` component.
```
```suggestion
Toolbars consist of content rows of components. Most toolbars will only be a single row, but each row needs to be housed in a `DataToolbarContent` component.
```

abigaeljamie commented on react-components/toolbar-filter/step1.md:
```diff
@@ -1,4 +1,69 @@
+&lt;/DataToolbar&gt;
+</pre>
+
+Once the preview reloads - it should look like this:
```
```suggestion
Once the preview reloads, it should look like this:
```

abigaeljamie commented on react-components/toolbar-filter/step2.md:
```diff
@@ -1,61 +1,141 @@
-      &lt;/React.Fragment&gt;
-    );
-  }
+The individual components placed inside of a toolbar are each wrapped in a `DataToolbarItem` component. `DataToolbarItems` can be passed an optional `variant` prop to format specific type of component it wraps such as a search filter, a pagination component, a bulk select, or an overflow menu. 
```
```suggestion
The individual components placed inside of a toolbar are each wrapped in a `DataToolbarItem` component. `DataToolbarItems` can be passed an optional `variant` prop to format specific a type of component it wraps such as a search filter, a pagination component, a bulk select, or an overflow menu. 
```

abigaeljamie commented on react-components/toolbar-filter/step2.md:
```diff
@@ -1,61 +1,141 @@
-  }
+The individual components placed inside of a toolbar are each wrapped in a `DataToolbarItem` component. `DataToolbarItems` can be passed an optional `variant` prop to format specific type of component it wraps such as a search filter, a pagination component, a bulk select, or an overflow menu. 
+
+By wrapping components in the `DataToolbarItem` component and adding the proper variant, it ensures the components are spaced appropriately.
```
```suggestion
Wrapping components in the `DataToolbarItem` component and adding the proper variant ensures the components are spaced appropriately.
```

abigaeljamie commented on react-components/toolbar-filter/step6.md:
```diff
@@ -1,95 +1,13 @@
-One particular type of Data toolbar item is called the `DataToolbarFilter`. When used, selected filters are displayed in chip groups below the toolbar. You may consider placing any `DataToolbarFilter` inside of a toggle group to make your toolbar responsive.
+Managing the relationship of a toolbar with its filters to a table or other component is the responsibility of the consumer. 
```
Who do we mean by consumer?

nicolethoen replied:
We mean the person trying to use the component in their code. Not the end user.

gheake2 replied:
I believe the user/developer building the toolbar.

abigaeljamie commented on react-components/toolbar-filter/step6.md:
```diff
@@ -1,95 +1,13 @@
+The `DemoTable` component is defined in `src/components/table.js` where the managed list of applied filters manipulates a very small set of data.
 
-At smaller viewport widths, you should see a responsive collapsed view of the toolbar and applied filters.
+Once the preview reloads - it should look like this:
```
```suggestion
Once the preview reloads, it should look like this:
```

gheake2 commented on react-components/toolbar-filter/index.json:
```diff
@@ -10,32 +10,28 @@
     "steps": [
       {
-        "title": "Introducing Patternfly React toolbar",
+        "title": "Set up DataToolbar basics",
```
```suggestion
        "title": "Setting up `DataToolbar` basics",
```

gheake2 commented on react-components/toolbar-filter/step1.md:
```diff
@@ -1,4 +1,69 @@
+
+2) **Set up the general structure of an empty toolbar and add the necessary imports**
+
+Copy the following code into the App.js file, replacing all of the content there:
```
```suggestion
Copy the following code into the `App.js` file, replacing all of the content there:
```

gheake2 commented on react-components/toolbar-filter/assets/table.js:
```diff
@@ -16,7 +16,7 @@ import {
 import { SearchIcon } from '@patternfly/react-icons';
 
-class AppTable extends React.Component {
+class DemoTable extends React.Component {
```
```suggestion
class `DemoTable` extends `React.Component {`
```

nicolethoen replied:
this is actual JS code that I need to run. I cannot add the ticks.

gheake2 commented on react-components/toolbar-filter/assets/table.js:
```diff
@@ -94,4 +94,4 @@ class AppTable extends React.Component {
 }
 
-export default AppTable;
+export default DemoTable;
```
```suggestion
export default `DemoTable`;
```

nicolethoen replied:
same comment here.

nicolethoen replied:
The ticks can only be added to files that have a .md extension

gheake2 replied:
got it..thanks

christiemolloy commented on react-components/toolbar-filter/step1.md:
```diff
@@ -1,4 +1,69 @@
+
+The `DataToolbar` should include the property `id="tutorial-toolbar"`.
+
+4) **Wrap the `Toolbar Demo` text inside `DataToolbar` in a `DataToolbarContent` component**
```
Now that the text has been wrapped in a DataToolbar component, I think you can change this step to be:

wrap the DataToolbar component added in step 3 in a DataToolbarContent component.

nicolethoen replied:
hm... except it goes the other way around. We need them to wrap the TEXT that's already in the DataToolbar component in a DataToolbarContent component. DataToolbar is the outer most component inside the React.Fragment.

And this is hard to explain in steps, but this process is how the developer will build their toolbar. They build the DataToolbar, and then place DataTooblarContents as rows within it. 

What makes it confusing is the addition of the text 'Toolbar Demo'. But without that text, there would be no feedback to the user that anything is working. The preview screen will be empty. That's why I start with the text in the code. 

I could reorder the steps so that the text is wrapped in the DataToolbarContent first, and then the DataToolbarContent is wrapped in the DataToolbar, but then they will see an error in the preview between steps and this will not be natural development flow.

I think the best solution is to remove the 'Toolbar Demo' text so they are just putting components within components without having to wrap things in this sort of confusing manner. The only downside to this, there will be no preview for this step.

christiemolloy commented on react-components/toolbar-filter/step2.md:
```diff
@@ -1,61 +1,141 @@
-      &lt;/React.Fragment&gt;
-    );
-  }
+The individual components placed inside of a toolbar are each wrapped in a `DataToolbarItem` component. `DataToolbarItems` can be passed an optional `variant` prop to format a specific type of component it wraps such as a search filter, a pagination component, a bulk select, or an overflow menu. 
```
```suggestion
The individual components placed inside of a toolbar are each wrapped in a `DataToolbarItem` component. `DataToolbarItems` can be passed an optional `variant` prop to format a specific type of component that it. `DataToolbarItems` wrap components like the search filter, pagination component, bulk select, or overflow menu. 
```

nicolethoen replied:
The grammar in this suggestion seems off.
I'll change it to: The individual components placed inside of a toolbar are each wrapped in a `DataToolbarItem` component. `DataToolbarItems` can be passed an optional `variant` prop to format a specific type of component it wraps. These specific components could be search filter, a pagination component, a bulk select, or an overflow menu.

christiemolloy commented on react-components/toolbar-filter/step2.md:
```diff
@@ -1,61 +1,141 @@
+
+Use the state of the `ToolbarDemoApp` to manage the three applied filters in the `DataToolbar`.
+
+1) **Locate the following constructor in `ToolbarDemoApp**
```
```suggestion
1) **Locate the following constructor in `ToolbarDemoApp`**
```

christiemolloy commented on react-components/toolbar-filter/step2.md:
```diff
@@ -1,61 +1,141 @@
+
+2) **Initialize the state in the constructor located in step 1**
+
+Add a `filters` object to the state. `filters` should have three properties (name, risk, and status) each with an empty array as its value. This filters object will manage the three filters in the `DataToolbar`.
```
```suggestion
Add a `filters` object to the state. `filters` should have three properties (name, risk, and status) each with an empty array as their value. This filters object will manage the three filters in the `DataToolbar`.
```

nicolethoen replied:
The subject of this phrase is each [one] and "value" is singular, so the possessive pronoun should be singular. 

I can clarify and rewrite the sentence to read: "`filters` should have three properties (name, risk, and status), and each property should have an empty array as its value."

christiemolloy commented on react-components/toolbar-filter/step2.md:
```diff
@@ -1,61 +1,141 @@
+
+Both of the event handlers can be inserted outside and below the constructor modified in the previous steps.
+
+The first will be an `onSearch` event handler which will fire when the user interacts with the name filter.
```
Make this a little more actionable:

```suggestion
The first event handler to add is `onSearch` which will fire when the user interacts with the name filter.
```

christiemolloy commented on react-components/toolbar-filter/step2.md:
```diff
@@ -1,61 +1,141 @@
+}
+</pre>
+
+The second event handler will be an `onSelect` event handler which will fire when the user interacts with the risk and status filters.
```
```suggestion
The second event handler to add is `onSelect` which will fire when the user interacts with the risk and status filters.
```

christiemolloy commented on react-components/toolbar-filter/step2.md:
```diff
@@ -1,61 +1,141 @@
 };
+</pre>
+
+4) **Add the following constants inside the top of the render function above the return statement**
```
This has a lot of instructions before introducing what the purpose of this step is. Change to something like:
`Declare constants in the render function.`

And then specify the location

christiemolloy commented on react-components/toolbar-filter/step2.md:
```diff
@@ -1,61 +1,141 @@
+&lt;/DataToolbar&gt;
+</pre>
+
+6) **Replace the `Toolbar Demo` text in the `DataToolbarContent` with the following six items:**
```
I think you should add a sentence here that explains briefly what six items are being added

christiemolloy commented on react-components/toolbar-filter/step3.md:
```diff
@@ -1,131 +1,106 @@
-    };
-  }
-</pre>
+2) **Wrap both the `DataToolbarItem` components containing the `CheckboxSelect` filters in the same `DataToolbarGroup` component**
```
Its a little confusing as to whether to wrap both components in DataToolbarGroup or use one wrapper around both.

```suggestion
2) **Wrap both the `DataToolbarItem` components containing the `CheckboxSelect` filters in one `DataToolbarGroup` component**
```

christiemolloy commented on react-components/toolbar-filter/step3.md:
```diff
@@ -1,131 +1,106 @@
-    });
-  };
-</pre>
+3) **Add the prop `variant="filter-group"` to the newly added `DataToolbarGroup`**
```
```suggestion
3) **Add the prop `variant="filter-group"` to the `DataToolbarGroup` added in step 2**
```

christiemolloy commented on react-components/toolbar-filter/step3.md:
```diff
@@ -1,131 +1,106 @@
+&lt;/DataToolbarItem&gt;
 </pre>
 
+5) **Wrap three adjacent `DataToolbarItem` components containing icons in the same `DataToolbarGroup` component**
```
```suggestion
5) **Wrap the three adjacent `DataToolbarItem` components containing icons in one `DataToolbarGroup` component**
```

christiemolloy commented on react-components/toolbar-filter/step3.md:
```diff
@@ -1,131 +1,106 @@
+5) **Wrap three adjacent `DataToolbarItem` components containing icons in the same `DataToolbarGroup` component**
 
-6) Locate the code that looks like the following:
+6) **Add the prop `variant="icon-button-group"` to the most recently added `DataToolbarGroup`**
```
```suggestion
6) **Add the prop `variant="icon-button-group"` to the `DataToolbarGroup` component added in step 5**
```

christiemolloy commented on react-components/toolbar-filter/step4.md:
```diff
@@ -1,29 +1,53 @@
-Toolbar groups are used to group sets of like items to create desired associations and to enable items to respond together to changes in viewport width. A `DataToolbarGroup` is passed an optional prop `variant` to format particular types of groups of elements such as filter groups, button groups, or icon button groups.
+One particular type of Data toolbar group is called the toggle group. A `DataToolbarToggleGroup` can be used to collapse a set of items into an overlay panel at a certain breakpoint. This allows complex toolbars with multiple items and groups of items to be responsive. 
```
sentence case for data toolbar

christiemolloy commented on react-components/toolbar-filter/step4.md:
```diff
@@ -1,29 +1,53 @@
+
+Using a `DataToolbarToggleGroup` requires both a `toggleIcon` and `breakpoint` property. The `toggleIcon` is a ReactNode which is displayed instead of the children of `DataToolbarToggleGroup` once the viewport width shrinks below the passed in breakpoint.
+
+1) **Locate both the `DataToolbarItem` with `variant="search-filter"` and the `DataToolbarGroup` with `variant="filter-group"`:**
```
```suggestion
1) **Locate both the `DataToolbarItem` with `variant="search-filter"` and the `DataToolbarGroup` with `variant="filter-group"`.**
```

christiemolloy commented on react-components/toolbar-filter/step4.md:
```diff
@@ -1,29 +1,53 @@
 </pre>
 
-2) Wrap both the `DataToolbarItem` components containing our `CheckboxSelect` filters in the same `DataToolbarGroup` along with the optional prop `variant="filter-group"`.
+2) **Wrap both the located `DataToolbarItem` and `DataToolbarGroup` in the same `DataToolbarToggleGroup`**
```
```suggestion
2) **Wrap the `DataToolbarItem` and `DataToolbarGroup` located in step 1, in a `DataToolbarToggleGroup`**
```

christiemolloy commented on react-components/toolbar-filter/step4.md:
```diff
@@ -1,29 +1,53 @@
+2) **Wrap both the located `DataToolbarItem` and `DataToolbarGroup` in the same `DataToolbarToggleGroup`**
 
-<pre class="file" data-target="clipboard">
+3) **Add the `toggleIcon` property to the `DataToolbarToggleGroup` component. Its value should be `{<FilterIcon />}`**
```
Split this step up, and move the second sentence to a new line.

```suggestion
3) **Add the `toggleIcon` property to the `DataToolbarToggleGroup` component. **

The value of the `toggleIcon` property should be `{<FilterIcon />}`
```

christiemolloy commented on react-components/toolbar-filter/step4.md:
```diff
@@ -1,29 +1,53 @@
-<pre class="file" data-target="clipboard">
+3) **Add the `toggleIcon` property to the `DataToolbarToggleGroup` component. Its value should be `{<FilterIcon />}`**
+
+4) **Add the `breakpoint` property to the `DataToolbarToggleGroup` component. Its value should be `"xl"`**  
```
```suggestion
4) **Add the `breakpoint` property to the `DataToolbarToggleGroup` component.** 

The value of the `breakpoint` property should be `"xl"`  
```

christiemolloy commented on react-components/toolbar-filter/step4.md:
```diff
@@ -42,50 +66,21 @@ Toolbar groups are used to group sets of like items to create desired associatio
-    &lt;/Button&gt;
-  &lt;/DataToolbarItem&gt;
-</pre>
+5) **Test changing viewport widths and breakpoints**
```
```suggestion
5) **Test the responsiveness of the toolbar**
```

christiemolloy commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,120 @@
-One particular type of Data toolbar group is called the toggle group. A `DataToolbarToggleGroup` can be used when you want to collapse a set of items into an overlay panel at a certain breakpoint. This allows complex toolbars with multiple items and groups of items to be responsive. A toggle group is useful for containing filter controls, for example. When the toolbar responds to adapt to a mobil ...
+One particular type of Data toolbar item is called the `DataToolbarFilter`. When used, selected filters are displayed in chip groups below the toolbar. Placing any `DataToolbarFilter` inside of a toggle group makes the toolbar responsive.
```
lowercase for data

christiemolloy commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,120 @@
 
-## Task
-1) Locate the following "search-filter" `DataToolbarItem` and the "filter-group" `DataToolbarGroup`:
+1) **Define the required `onDelete` function to be passed to the DataToolbarFilters**
```
```suggestion
1) **Define the required `onDelete` function to be passed to the `DataToolbarFilters`**
```

christiemolloy commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,120 @@
-1) Locate the following "search-filter" `DataToolbarItem` and the "filter-group" `DataToolbarGroup`:
+1) **Define the required `onDelete` function to be passed to the DataToolbarFilters**
+
+Add it to the ToolbarDemoApp class below the `onSelect` function definition in the `ToolbarDemoApp` class.
```
```suggestion
Add the function to the ToolbarDemoApp class. Make sure to add it below the `onSelect` function definition in the `ToolbarDemoApp` class.
```

christiemolloy commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,120 @@
 </pre>
 
-2) Wrap both the `DataToolbarItem` and the `DataToolbarGroup` in the same `DataToolbarToggleGroup`. Provide both the `toggleIcon` and `breakpoint` properties to the `DataToolbarToggleGroup`.
+3) **Replace each of the three `DataToolbarItems` inside the `DataToolbarToggleGroup` with a `DataToolbarFilter`**
```
This has potential to confuse the user, add an example of what it should look like at this step

nicolethoen replied:
hm.... there will be no visible change...

nicolethoen replied:
oh I see, like a code snippet. okay

christiemolloy commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,120 @@
-2) Wrap both the `DataToolbarItem` and the `DataToolbarGroup` in the same `DataToolbarToggleGroup`. Provide both the `toggleIcon` and `breakpoint` properties to the `DataToolbarToggleGroup`.
+3) **Replace each of the three `DataToolbarItems` inside the `DataToolbarToggleGroup` with a `DataToolbarFilter`**
+
+4) **Add a `chips` property to each of the `DataToolbarFilter`**
```
```suggestion
4) **Add a `chips` property to each of the `DataToolbarFilter` components.**
```

christiemolloy commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,120 @@
+
+Each of the `DataToolbarFilter` components requires a `chips` property. In this case, the state managed filters array's properties can serve as the `chips` property for each `DataToolbarFilter`. 
+
+* In this case, the first `DataToolbarFilter` (containing a text input) should have the property `chips={filters.name}`. 
```
Since these are actions, I think put them in a) b) c)

christiemolloy commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,120 @@
+
+5) **Pass an appropriate label as the `categoryName` property to each `DataToolbarFilter`**
+
+* In this case, the first `DataToolbarFilter` (containing a text input) should have the property `categoryName="Name"`. 
```
Same comment as above, a) b) c)

christiemolloy commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,120 @@
+* The second `DataToolbarFilter` (containing the status filter) should have the property `categoryName="Status"`. 
+* The third `DataToolbarFilter` (containing the risk filter) should have the property `categoryName="Risk"`.
+
+6) **Pass the onDelete function as the `deleteChip` property to each `DataToolbarFilter`**
```
Give an example here of what this looks like:

<DataToolbarFilter deleteChip={this.onDelete}>

nicolethoen replied:
I have a code snippet after step 6. I will reword the next line to say 'Once steps 4, 5, and 6 are completed, the contents of the `DataToolbarToggleGroup` should appear as follows:"

christiemolloy commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,120 @@
+7) **Experiment with two optional properties passed to `DataToolbar`**
+
+Two props can be optionally added to the top level `DataToolbar` component when using a `DataToolbarFilter`.
+* If a `clearAllFilters` event handler is passed to `DataToolbar`, a 'Clear All Filters' action will appear alongside the applied filters chip groups.
```
Put these in a) and b)

christiemolloy commented on react-components/toolbar-filter/step6.md:
```diff
@@ -1,95 +1,13 @@
-</pre>
-
-3) Replace each of the three `DataToolbarItems` inside the `DataToolbarToggleGroup` with a `DataToolbarFilter`. Pass the appropriate state managed filters array as the `chips` property, an appropriate label as the `categoryName` property, and the onDelete function as the `deleteChip` property. 
+1) Add the `DemoTable` component below the `DataToolbar` and pass it the status and risk filters as props.
```
Make this bold. ALso theres a lot going on in this step, so separate it out by first having the overall action of the step, and then putting the other information in its own sentence

christiemolloy commented on react-components/toolbar-filter/step6.md:
```diff
@@ -1,95 +1,13 @@
+</pre> 
 
-<img src="toolbar-filter/assets/wide-view-with-chip-groups.png" alt="Wide view of toolbar filter with applied filters in chip groups" style="box-shadow: rgba(3, 3, 3, 0.2) 0px 1.25px 2.5px 0px;" />
+The `DemoTable` component is defined in `src/components/table.js` where the managed list of applied filters manipulates a very small set of data.
```
Should this information be moved up underneath the step, or is this more of a "Note"

nicolethoen replied:
It's more of a note. I'll add **Note:** before it.

kmcfaul commented:
Couple items of feedback after running through the tutorial.

- On Step 5, No. 4 & No.5: the sub-steps (a, b, c, etc) are inline, making it a little hard to read.
- Also on Step 5, No. 7: I was playing around with the filters before adding in the handler for clearAllFilters, and the "Clear all filters" button appeared without the properties being passed in, so it wasn't functional at that point.
- Step 6: When the editor reloads after adding in the DataTable, the filters are reset, so the ending preview image is not accurate because it shows selected filters. It's easy to see how the image is achieved, but may still want a note stating which filters are selected to get the image.

gheake2 commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,146 @@
+b) The second `DataToolbarFilter` (containing the status filter) should have the property `categoryName="Status"`. 
+c) The third `DataToolbarFilter` (containing the risk filter) should have the property `categoryName="Risk"`.
+
+6) **Pass the onDelete function as the `deleteChip` property to each `DataToolbarFilter`**
```
```suggestion
6) <strong>Pass the onDelete function as the `deleteChip` property to each `DataToolbarFilter`</strong>
```

gheake2 commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,146 @@
+
+7) **Experiment with two optional properties passed to `DataToolbar`**
+
+Two props can be optionally added to the top level `DataToolbar` component when using a `DataToolbarFilter`.
```
```suggestion
Two props can be optionally added to the top-level `DataToolbar` component when using a `DataToolbarFilter`.
```

gheake2 commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,146 @@
-<img src="toolbar-filter/assets/toggle-group-wide-view.png" alt="wide view of toggle group" style="box-shadow: rgba(3, 3, 3, 0.2) 0px 1.25px 2.5px 0px;" />
-This second image shows how the toolbar should appear at a narrow viewport width.
-<img src="toolbar-filter/assets/toggle-group-narrow-view.png" alt="narrow view of toggle group" style="box-shadow: rgba(3, 3, 3, 0.2) 0px 1.25px 2.5px 0px;" />
+Take some time to play with the toolbar to see how it responds to changes in viewport size and the number of filters applied.
```
```suggestion
<strong> Note:</strong>Take some time to play with the toolbar to see how it responds to changes in viewport size and the number of filters applied.
```

gheake2 commented on react-components/toolbar-filter/step6.md:
```diff
@@ -1,95 +1,15 @@
+Managing the relationship of a toolbar with its filters to a table or other component is the responsibility of the consumer. 
 
-Using a `DataToolbarFilter` requires three properties. First is a managed array of selected filters called `chips` as strings or ReactNodes. It also requires an `onDelete` callback function to be executed whenever the user deletes a selected filter chip. Lastly, it requires a `categoryName` which will be used to label the chip group.
+1) **Add the `DemoTable` component below the `DataToolbar`** 
```
```suggestion
1) <strong>Add the `DemoTable` component below the `DataToolbar`</strong> 
```

nicolethoen commented on react-components/toolbar-filter/step3.md:
```diff
@@ -1,131 +1,106 @@
-The individual components placed inside of a toolbar are each wrapped in a `DataToolbarItem` component. `DataToolbarItems` can be passed an optional `variant` prop to format specific type of component it wraps such as a search filter, a pagination component, a bulk select, or an overflow menu. By wrapping components in the `DataToolbarItem` component and adding the proper variant, you are ensurin ...
+Toolbar groups are used to group sets of like items to create desired associations and to enable items to respond together to changes in viewport width. A `DataToolbarGroup` is passed an optional prop `variant` to format particular types of groups of elements such as filter groups, button groups, or icon button groups.
```
um... the concept of Toolbar groups is not introduced until this step. They are a subset of Toolbars and I assumed the intro would be more of a place to introduce the concept and purpose of the overall Toolbar, not a small aspect of the toolbar such as the toolbar groups.

nicolethoen commented:
> * Also on Step 5, No. 7: I was playing around with the filters before adding in the handler for clearAllFilters, and the "Clear all filters" button appeared without the properties being passed in, so it wasn't functional at that point.

@kmcfaul I addressed your other comments, but this issue is something that I fixed in React with PR https://github.com/patternfly/patternfly-react/pull/3398. if you wouldn't mind reviewing that. When it goes in, then the clear all filters button won't appear until an event handler is provided.

christiemolloy commented on react-components/toolbar-filter/step5.md:
```diff
@@ -1,67 +1,152 @@
+
+a) If a `clearAllFilters` event handler is passed to `DataToolbar`, a 'Clear All Filters' action will appear alongside the applied filters chip groups.
+
+b) If a `collapseListedFiltersBreakpoint` is passed to `DataToolbar`, the chip groups will collapse to a summary message at the passed in breakpoint, making the applied filters chip groups more responsive.
```
```suggestion
b) <strong>Pass a`collapseListedFiltersBreakpoint` to `DataToolbar`.</strong>

The chip groups will collapse to a summary message at the passed in breakpoint, making the applied filters chip groups more responsive.
```
